source/adapters/level_zero/queue.hpp

```diff
@@ -234,6 +234,9 @@ struct ur_queue_handle_t_ : _ur_object {
   // Keeps the properties of this queue.
   ur_queue_flags_t Properties;
 
+  // Keeps track of whether we are using Counter-based Events
+  bool counterBasedEventsEnabled = false;
```

Comment:
nit: variable naming not quite consistent
```suggestion
  bool CounterBasedEventsEnabled = false;
```